### PR TITLE
Add connection status observer state and more log

### PR DESCRIPTION
**Pull request by Erli-ms**

### CODE REVIEW

File: src/enhancers/exportDLJSInterface.ts [hunk 1]
```diff
@@ -48,8 +49,14 @@ export default function exportDLJSInterface<TAdapterState extends AdapterState>(
 
     adapter.addEventListener('open', async () => {
       if(!connectionStatusObserver){
-        let waitTime = 5;
-        while(!connectionStatusObserver && waitTime < 2000){
+        adapter.getState(StateKey.Logger)?.logClientSdkTelemetryEvent(Microsoft.CRM.Omnichannel.IC3Client.Model.LogLevel.ERROR,
```
Comment (charliewang95) [inline src/enhancers/exportDLJSInterface.ts:52]: Is this supposed to be Error? If this is possible/not fatal should we change it to WARN?

File: src/enhancers/exportDLJSInterface.ts [hunk 2]
```diff
@@ -85,12 +102,25 @@ export default function exportDLJSInterface<TAdapterState extends AdapterState>(
           (async function () {
             try {
               for await (const activity of adapter.activities({ signal: abortController.signal })) {
+                (window as any).ic3Logger?.logClientSdkTelemetryEvent(Microsoft.CRM.Omnichannel.IC3Client.Model.LogLevel.DEBUG,
```
Comment (charliewang95) [inline src/enhancers/exportDLJSInterface.ts:105]: why using window variable here?

File: src/ic3/enhancers/ingress/subscribeNewMessageAndThreadUpdate.ts [hunk 3]
```diff
@@ -71,11 +72,20 @@ export default function createSubscribeNewMessageAndThreadUpdateEnhancer(): Adap
...
+                while((getReadyState() != ReadyState.OPEN || !getState(StateKey.ConnectionStatusObserverReady)) && waitTime <= 2048){
                   await timeout(waitTime);
                   waitTime *= 2;
                 }
+                //log only 
+                if(waitTime > 2) {
+                  getState(StateKey.Logger)?.logClientSdkTelemetryEvent(Microsoft.CRM.Omnichannel.IC3Client.Model.LogLevel.INFO,
+                    {
+                      Event: TelemetryEvents.ADAPTER_STATE_UPDATE,
+                      Description: `Adapter: connectionStatusObserverReady state after waitting for ${waitTime} ms: `+getState(StateKey.ConnectionStatusObserverReady)
```
Comment (charliewang95) [inline src/ic3/enhancers/ingress/subscribeNewMessageAndThreadUpdate.ts:85]: typo

File: src/utils/logMessageFilter.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import { ActivityType, IDirectLineActivity, Role, TwoWaySerializableComplex } from '../types/DirectLineTypes';
```
Comment (charliewang95) [inline src/utils/logMessageFilter.ts:1]: Keep the same indent for this file (2)?

File: src/utils/logMessageFilter.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    textLength?: number;
+    id?: string;
+    suggestedActionCount?: number;
+    timestamp: string;
+    type: ActivityType;
+    hasValue?: boolean;
+    messageid?: string;
+    previousClientActivityID?: string;
+}
+export function logMessagefilter(message: IDirectLineActivity): string {
```
Comment (charliewang95) [inline src/utils/logMessageFilter.ts:22]: I fear this is going to overload the database...

File: src/enhancers/exportDLJSInterface.ts (hunk 1 above)
Comment (Erli-ms) [reply src/enhancers/exportDLJSInterface.ts:52]: good catch.

File: src/enhancers/exportDLJSInterface.ts (hunk 2 above)
Comment (Erli-ms) [reply src/enhancers/exportDLJSInterface.ts:105]: fixed

File: src/ic3/enhancers/ingress/subscribeNewMessageAndThreadUpdate.ts (hunk 3 above)
Comment (Erli-ms) [reply src/ic3/enhancers/ingress/subscribeNewMessageAndThreadUpdate.ts:85]: good catch.

File: src/utils/logMessageFilter.ts (hunk 5 above)
Comment (Erli-ms) [reply src/utils/logMessageFilter.ts:22]: Should be OK. compared to the number of notification updating messages, this is only a small piece.

File: src/utils/logMessageFilter.ts (hunk 4 above)
Comment (Erli-ms) [reply src/utils/logMessageFilter.ts:1]: fixed

**Pull request merged**